### PR TITLE
add reading progress bar to the Navbar

### PR DESCRIPTION
**Pull request by garfieldduck**
- set Nabber's position to fixed
- add redux store to keep current scrolling position
- remove react-sticky on mobile navbar

### CODE REVIEW

File: src/components/HeaderProgress.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        </div>
+      </div>
+    )
+  }
+}
+
+function mapStateToProps(state) {
+  return {
+    header: state.header || {},
+    entities: state.entities || {}
```
Comment (hcchien) [inline src/components/HeaderProgress.js]: I guess we don't need the entities here?

File: src/components/HeaderProgress.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+'use strict'
+import { connect } from 'react-redux'
+import React, { Component } from 'react' // eslint-disable-line
+import styles from './HeaderProgress.scss'
+import { setReadProgress } from '../actions/header'
```
Comment (hcchien) [inline src/components/HeaderProgress.js]: make some readable sorting? :)

File: src/components/NavBar.js [hunk 3]
```diff
@@ -1,7 +1,12 @@
+import _ from 'lodash'
 import React, { Component, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
 import DesktopNavBar from './DesktopNavBar'
+import HeaderProgress from './HeaderProgress'
 import MobileNavBar from './MobileNavBar'
```
Comment (hcchien) [inline src/components/NavBar.js]: import sorting

File: src/components/HeaderProgress.js (hunk 2 above)
Comment (nickhsine) [inline src/components/HeaderProgress.js]: Do we need ```setReadProgress``` here?

File: src/components/HeaderProgress.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+function mapStateToProps(state) {
+  return {
+    header: state.header || {},
+    entities: state.entities || {}
+  }
+}
+
+export default connect(mapStateToProps, { setReadProgress })(HeaderProgress)
```
Comment (nickhsine) [inline src/components/HeaderProgress.js]: seems that we don't need ```{ setReadProgress } ``` here?

File: src/components/MobileNavBar.js [hunk 5]
```diff
@@ -62,24 +62,22 @@ export default class MobileNaviBar extends Component {
...
-            <span></span>
-            <span></span>
-            <span></span>
-            <span></span>
-          </div>
-          <div className={mobileMenuClass}>
-            {navLinks}
-            <SubNavBar {...this.props}/>
-          </div>
+      <div className={mobileNavClass} style= {{ backgroundColor: backgroundColor }}>
```
Comment (nickhsine) [inline src/components/MobileNavBar.js]: remove ```import { Sticky } from 'react-sticky'``` 
and update package.json?

File: src/components/NavBar.js [hunk 6]
```diff
@@ -10,42 +15,64 @@ if (process.env.BROWSER) {
 export default class NaviBar extends Component {
   constructor(props) {
     super(props)
+    this.state = {
+      height: 10
```
Comment (nickhsine) [inline src/components/NavBar.js]: use ```DEFAULT_HEIGHT``` ?

File: src/containers/Article.js [hunk 7]
```diff
@@ -41,6 +64,33 @@ export default class Article extends Component {
     return true
   }
 
+  _setArticleBounding() {
+    const top = ReactDOM.findDOMNode(this.refs.progressBegin).offsetTop
+    const bottom = ReactDOM.findDOMNode(this.refs.progressEnding).offsetTop
+    if(this.state.topY !== top || this.state.bottomY !== bottom) {
+      this.setState({
+        topY: top,
+        bottomY: bottom
```
Comment (nickhsine) [inline src/containers/Article.js]: could we just use global variable to store ```topY``` and ```bottomY``` ?

**Comment by nickhsine** [comment]
💯  ㄅ 棒

**Comment by nickhsine** [comment]
五個燈 💡 💡 💡 💡 💡

**Pull request merged**